File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indices = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            int val = target - nums[i];
            if (indices.containsKey(val)) {
                return new int[]{indices.get(val), i};
            }
            if (!indices.containsKey(nums[i])) {
                indices.put(nums[i], i);
            }
        }

        return new int[]{};
    }
}

File: Data Structures & Algorithms/can-place-flowers/submission-1.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        nums = [0] + flowerbed + [0]

        for i in range(1, len(nums) - 1):
            if (nums[i] == 0 and nums[i - 1] == 0 and nums[i + 1] == 0):
                n -= 1
                nums[i] = 1

        return n <= 0
